includes/User.php

```diff
@@ -592,6 +592,40 @@ public static function whoIs( $id ) {
 	}
 
 	/**
+	 *
+	 * @param $ids Array User IDs
+	 * @return Array User ID to User name mapping
+	 */
+	public static function whoAre( Array $ids, $source = DB_SLAVE ): Array {
+		global $wgExternalSharedDB;
+		$sdb = wfGetDB( $source, [], $wgExternalSharedDB );
+		$res = $sdb->select(
+			'`user`',
+			[ 'DISTINCT user_id', 'user_name' ],
+			[ 'user_id' => $ids ],
+			__METHOD__
+		);
+
+		// Pre-fill the returned array with empty strings
+		// so that missing users are not skipped.
+		// It makes further iterating over the array
+		// and handling anons and missing users a little
+		// bit easier.
+		$users = array_fill_keys( $ids, '' );
```

Comment:
Should we log missing user names?

Reply:
I don't think so.  I would leave it to the programmer using this method to deal with the output properly.  The sole purpose of this mapping is to provide a convenient mapping.  We can call this method within some other method created for data integrity checks.